src/bll/weblinklistener/httphostrequest.ts

```diff
@@ -0,0 +1,72 @@
+import {HttpRequestData} from "./httprequestdata";
+import {WorkspaceProxy} from "../moduleproxies/workspace-proxy";
+import * as path from "path";
+import {FsProxy} from "../moduleproxies/fs-proxy";
+import {Logger} from "../utils/logger";
+import {inject, injectable} from "inversify";
+import {TYPES} from "../utils/constants";
+import {WindowProxy} from "../moduleproxies/window-proxy";
+import {UriProxy} from "../moduleproxies/uri-proxy";
+import {Utils} from "../utils/utils";
+import {parse as parseUrl} from "url";
+
+@injectable()
+export class HttpHostRequest {
+
+    public constructor(@inject(TYPES.WorkspaceProxy) private workspaceProxy: WorkspaceProxy,
+                       @inject(TYPES.FsProxy) private fsProxy: FsProxy,
+                       @inject(TYPES.UriProxy) private uriProxy: UriProxy,
+                       @inject(TYPES.WindowProxy) private windowProxy: WindowProxy) {
+    }
+
+    public async processRequest(request: {url?: string, method?: string}): Promise<boolean> {
+        if (!request.url) {
+            Logger.logWarning("HttpHostRequest#processRequest url is required for processing request");
+            return false;
+        }
+
+        let path = request.url;
+        let params = {};
+
+        const endOfPathIndex: number = request.url.indexOf("?");
```

Comment:
It looks like it could be replaced like that:
```js
let url = parseUrl(request.url, true);
return this.fireRequestAccepted(url.pathname, url.query);
```